File: python/005.py
# -*- coding: utf-8 -*-
from utils import gcd, lcm

def solve(r):
    ans = 1
    for i in range(2, r + 1):
        print ans,
        ans = lcm(ans, i)
        print i, ans
    return ans


if __name__ == '__main__':
    assert solve(10) == 2520
    assert solve(20) == 232792560
    #assert solve(30) == 2329089562800

File: ruby/005.rb
def gcd(x, y)
    if y == 0
        x
    else
        gcd(y, x % y)
    end
end


def lcm(x, y)
    x / gcd(x, y) * y
end


def solve(r)
    ans = 1
    for i in 2..r
        ans = lcm(ans, i)
    end
    ans
end


require 'test/unit'
class TestSolve < Test::Unit::TestCase
    def test_nums
        assert solve(10) == 2520
        assert solve(20) == 232792560
        assert solve(30) == 2329089562800
    end
end
